pkg/parser/epub/metadata.go

```diff
@@ -378,21 +378,66 @@ func (m MetadataParser) computeMetaItem(expr MetadataItem, metas map[string][]Me
 type metadataAdapter struct {
 	epubVersion float64
 	items       map[string][]MetadataItem
+	links       []EPUBLink
 }
 
 func (m metadataAdapter) Duration() *float64 {
 	return ParseClockValue(m.FirstValue(VocabularyMedia + "duration"))
 }
 
+func (m metadataAdapter) First(property string) (item MetadataItem, ok bool) {
+	items, ok := m.items[property]
+	if !ok || len(items) == 0 {
+		return
+	}
+	item = items[0]
+	return
+}
+
 func (m metadataAdapter) FirstValue(property string) string {
-	p, ok := m.items[property]
+	item, ok := m.First(property)
 	if !ok {
 		return ""
 	}
-	if len(p) == 0 {
-		return ""
+	return item.value
+}
+
+func (m metadataAdapter) Values(property string) []string {
+	values := []string{}
+	if items, ok := m.items[property]; ok {
+		for _, item := range items {
+			values = append(values, item.value)
+		}
+	}
+	return values
+}
+
+func (m metadataAdapter) Links(rel string) []EPUBLink {
+	links := []EPUBLink{}
```

Comment:
For this function, because we can't predict the size very well, it's fine just to let it grow with append so I don't suggest the same optimization.